### ISSUE TITLE
[BUG] forecasters broadcast predictions on wrong instances when label as str

### ISSUE THREAD

**Issue by hazrulakmal** [sktime/sktime#5108]
**Describe the bug**

When a panel datatype is given where the instances are labelled as str such that a random name followed by some int, e.g *user1*, *series1* , *T1* etc. the prediction that was made on the instances sometimes gets switched. below is a minimal example to show the problem. 

I suppose this occurs for all forecasters, albeit the test I did wasn't exhaustive, only on `TrendForecaster`, `NaiveForecaster` and `ExponentialSmothing`. I haven't looked at the logic of how forecasters made predictions for hierarchical or panel datatypes but I suspect they were all inherited from the same logic when making predictions for each single series. 

 

**To Reproduce**

```python
import pandas as pd
from sktime.forecasting.model_selection import ExpandingGreedySplitter
from sktime.forecasting.naive import NaiveForecaster

n_instances=3
n_points=5

temp = pd.DataFrame({
    'y': [i for i in range(n_points*n_instances)],
    'id':["T1"]*n_points + ["T2"]*n_points+ ["T11"]*n_points,  
    # bug still occurs even when timestamp is int-like
    'timestamp': pd.date_range(start='2022-07-01', periods=n_points*n_instances, freq='D') 
}).set_index(['id', 'timestamp'])

fh = [1,2]
train_test_splitter = ExpandingGreedySplitter(test_size=len(fh), folds=1)
forecaster = NaiveForecaster(strategy="last")  # NaiveForecaster() or TrendForecaster()
for train, test in train_test_splitter.split_series(temp): 
    forecaster.fit(train)
    y_pred = forecaster.predict(fh)
    print("=====TEST=====")
    print(test.head(15))
    print("=====PREDICT=====")
    print(y_pred.head(15))
    print("=====TRAIN=====")
    print(train.head(15))
```

**Output** - predictions for T2 and T11 got **switched** as shown in the output below
```
=====TEST=====
                 y
id  timestamp     
T1  2022-07-04   3
    2022-07-05   4
T2  2022-07-09   8
    2022-07-10   9
T11 2022-07-14  13
    2022-07-15  14
=====PREDICT=====
                   y
id  timestamp       
T1  2022-07-04   2.0
    2022-07-05   2.0
T2  2022-07-14  12.0  
    2022-07-15  12.0
T11 2022-07-09   7.0 
    2022-07-10   7.0 
=====TRAIN=====
                 y
id  timestamp     
T1  2022-07-01   0
    2022-07-02   1
    2022-07-03   2
T2  2022-07-06   5
    2022-07-07   6
    2022-07-08   7
T11 2022-07-11  10
    2022-07-12  11
    2022-07-13  12
```

**Expected behavior**

Trivially, the predictions should be broadcasted to the corresponding instance based on the trained data.

**Additional context**

This bug is spotted when trying to fit naive forecaster in M4 weekly dataset as per in this notebook #5051

**Versions**
<details>

sktime: 0.21.0
sklearn: 1.3.0
skbase: 0.4.6
numpy: 1.24.3
pandas: 2.0.3
</details>

@fkiraly @benHeid

**Comment by fkiraly** [comment]
Confirmed on current `main`, windows, python 3.11.

FYI @hoesler, seems to be the same bug as in #4683 - the index gets lexicographically ordered while the rows do not.

**Comment by fkiraly** [comment]
I think the reason is again `pandas` ordering lexicographically in a case where one wouldn't expect.

If you do `list(train.groupby(level=0))`, this orders lexicographically. This happens in `VectorizeDF.items`, in line 264. Same mistake (given unexpected behaviour of `pandas`)

**Comment by fkiraly** [comment]
Indeex, I can confirm that the problem is the same.

The fix is conceptually simple but tedious - you have to force `pandas` to not sort rows lexicographically in two places (not just one as in #4683).

Fix is in https://github.com/sktime/sktime/pull/5110.
